### PR TITLE
feat(dashboards) Attach widget validation endpoint up.

### PR DESCRIPTION
**Pull request by markstory**
Use the widget validation endpoint when adding widgets from the modal. If the request fails propagate errors to relevant inputs.

Screen Shot 2020-11-30 at 4 54 37 PM

### CODE REVIEW

**Review by wmak — APPROVED** [review]
I think we can do this in another PR, but the `get_filter` way of validating conditions needs the query params as well (eg. `project:foo` will fail even if `foo` is valid)

Otherwise this LGTM :+1:

**Review by k-fish — APPROVED** [review]

**Review by dashed — APPROVED** [review]

File: src/sentry/static/sentry/app/components/modals/addDashboardWidgetModal.tsx [hunk 1]
```diff
@@ -55,21 +67,52 @@ const newQuery = {
...
     event.preventDefault();
 
-    const {closeModal, onAddWidget} = this.props;
-    onAddWidget(this.state as Widget);
-    addSuccessMessage(t('Added widget.'));
-    closeModal();
+    const {api, closeModal, organization, onAddWidget} = this.props;
+    this.setState({loading: true});
+    try {
+      const widgetData: Widget = omit(this.state, ['errors', 'loading']);
```
Comment (Zylphrex) [inline src/sentry/static/sentry/app/components/modals/addDashboardWidgetModal.tsx:105]: Is there a reason to prefer `omit` over `pick` here? The [docs](https://lodash.com/docs/4.17.15#omit) mention that `omit` is considerably slower than `pick`.

**Review by Zylphrex — APPROVED** [review]

File: src/sentry/static/sentry/app/components/modals/addDashboardWidgetModal.tsx (hunk 1 above)
Comment (markstory) [reply src/sentry/static/sentry/app/components/modals/addDashboardWidgetModal.tsx:105]: No real reason. I can use pick instead.

**Comment by markstory** [comment]
> I think we can do this in another PR, but the get_filter way of validating conditions needs the query params as well (eg. project:foo will fail even if foo is valid)

@wmak I didn't consider the `project:x` text filter. I am mostly interested in preventing syntax errors. Perhaps using `parse_search_query` is a better fit for this. I can do another change to swap that in.

File: src/sentry/static/sentry/app/components/modals/addDashboardWidgetModal.tsx (hunk 1 above)
Comment (markstory) [reply src/sentry/static/sentry/app/components/modals/addDashboardWidgetModal.tsx:105]: Whoops, forgot to push the change with pick(). I'll do that in my next set of changes.

**Pull request merged**